src/compas/geometry/primitives/polyline.py

```diff
@@ -412,4 +412,74 @@ def divide_polyline_by_length(self, length, strict=True, tol=1e-06):
 
         return division_pts
 
+    def extend(self, length, both_ends=False):
+        """Extends a polyline from its start or end by a given length
+
+        Parameters:
+        -----------
+        length: list of float [start_offset, end_offset]
+
+        both_ends: bool
+            If True, both ends of Polyline will be extended
+
+        Returns
+        -------
+        Polyline
+        """
+        if not isinstance(length, list):
+            length = [length, length]
+
+        new_end_pt = self.points[-1] + self.lines[-1].vector.unitized().scaled(length[1])
+        self.points.append(new_end_pt)
+
+        if both_ends:
+            new_start_pt = self.points[0] + self.lines[0].vector.unitized().scaled(-length[0])
+            self.points.insert(0, new_start_pt)
+
+        return self
+
+    def shorten(self, length, both_ends=False):
+        """Shorten a polyline from start or end by a given length
+
+        Parameters:
+        -----------
+        length: float or list of float [start_offset, end_offset]
```

Comment:
same as above, I would rather be strict and stick to a single input type, if not, the syntax of `or` is with `|` so: `list[float, float] | float`